### PR TITLE
Add github_team_membership

### PR DESCRIPTION
**Pull request by wbeuil**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes/no
| 🚀 New feature?   | yes/no
| ⚠ Deprecations?   | yes/no
| ❌ BC Break       | yes/no
| 🔗 Related issues | #...
| ❓ Documentation  | yes 

## Description

@eliecharra I needed to change your ListTeamQuery schema since I need the slug string to query each member of a team.

### CODE REVIEW

**Review by lotoussa — COMMENTED** [review]
lgtm

File: pkg/remote/github/github_team_supplier_test.go [hunk 1]
```diff
@@ -26,18 +26,18 @@ func TestGithubTeamSupplier_Resources(t *testing.T) {
...
 		{
 			test:    "Multiple github teams with parent",
 			dirName: "github_teams_multiple",
 			mocks: func(client *MockGithubRepository) {
-				client.On("ListTeams").Return([]int{
-					4556811, // github_team.team1
-					4556812, // github_team.team2
-					4556814, // github_team.with_parent
+				client.On("ListTeams").Return([]Team{
+					{4556811, "team1"},       // github_team.team1
```
Comment (eliecharra) [inline pkg/remote/github/github_team_supplier_test.go:38]: The slug is useless there, we are not using it to read team you can remove it

File: pkg/remote/github/repository.go [hunk 2]
```diff
@@ -159,3 +170,56 @@ func (r githubRepository) ListTeams() ([]int, error) {
...
+		return nil, err
+	}
+
+	query := listTeamMembershipsQuery{}
+	results := make([]string, 0)
+	if r.config.Organization == "" {
+		return results, nil
+	}
+	variables := map[string]interface{}{
+		"cursor": (*githubv4.String)(nil),
```
Comment (eliecharra) [inline pkg/remote/github/repository.go:202]: You can remove cursor and slug init there they are useless

File: pkg/remote/github/repository.go [hunk 3]
```diff
@@ -159,3 +170,56 @@ func (r githubRepository) ListTeams() ([]int, error) {
...
+			if err != nil {
+				return nil, err
+			}
+			for _, membership := range query.Organization.Team.Members.Nodes {
+				results = append(results, fmt.Sprintf("%d:%s", team.DatabaseId, membership.Login))
+			}
+			if !query.Organization.Team.Members.PageInfo.HasNextPage {
+				break
+			}
+			variables["cursor"] = githubv4.NewString(query.Organization.Team.Members.PageInfo.EndCursor)
```
Comment (eliecharra) [inline pkg/remote/github/repository.go:221]: Can you try without `NewString` there ? It seems that we do not need a pointer for the cursor. I've used this for github_branch_protection but I didn't refactored other resources atm.

**Review by eliecharra — CHANGES_REQUESTED** [review]

File: pkg/remote/github/repository.go (hunk 3 above)
Comment (wbeuil) [reply pkg/remote/github/repository.go]: It works indeed

**Comment by eliecharra** [comment]
@wbeuil need a rebase and conflict resolution there

**Review by eliecharra — APPROVED** [review]

**Pull request merged**